iml-task-runner/src/stats.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::error::ImlTaskRunnerError;
+use bigdecimal::{BigDecimal, ToPrimitive};
+use iml_influx::{Client, Point, Points, Precision, Value};
+use iml_manager_client::Url;
+use iml_manager_env::{get_influxdb_addr, get_influxdb_metrics_db};
+use iml_postgres::sqlx::{self, PgPool};
+use iml_tracing::tracing;
+use std::time::Duration;
+use tokio::time;
+
+struct TaskStats {
+    actions: Vec<String>,
+    filesystem: String,
+    fids_total: Option<BigDecimal>,
+    fids_completed: Option<BigDecimal>,
+    fids_failed: Option<BigDecimal>,
+}
+
+const DELAY: Duration = Duration::from_secs(300);
+
+pub(crate) async fn collector(pool: PgPool) -> Result<(), ImlTaskRunnerError> {
+    let mut interval = time::interval(DELAY);
+
+    let influx_url: String = format!("http://{}", get_influxdb_addr());
+    tracing::debug!("influx_url: {}", &influx_url);
+
+    loop {
+        interval.tick().await;
+
+        let stats: Vec<TaskStats> = sqlx::query_as!(
+            TaskStats,
+            r#"SELECT SUM(fids_total) AS fids_total,
+SUM(fids_completed) AS fids_completed,
+SUM(fids_failed) AS fids_failed,
+actions,
+fs.name AS filesystem FROM chroma_core_task AS t
+JOIN chroma_core_managedfilesystem AS fs ON t.filesystem_id = fs.id GROUP BY actions,fs.name"#
```

Comment:
```suggestion
            r#"SELECT COALESCE(SUM(fids_total), 0) AS "fids_total!",
COALESCE(SUM(fids_completed)) AS "fids_completed!",
COALESCE(SUM(fids_failed)) AS "fids_failed!",
actions,
fs.name AS filesystem FROM chroma_core_task AS t
JOIN chroma_core_managedfilesystem AS fs ON t.filesystem_id = fs.id GROUP BY actions,fs.name"#
```

This forces null `SUM` fields to be 0.  `TaskStats` can be updated to remove all the sums.

Reply:
SQLX didn't seem to care about the COALESCE when I tried that.

Reply:
It's not the `COALESCE` part, it's the cast I added afterwards that allows this to compile.